core/server/api_container/server/startosis_engine/startosis_validator/validator_environment.go

```diff
@@ -60,6 +63,10 @@ func (environment *ValidatorEnvironment) AppendRequiredImageBuild(containerImage
 	environment.imagesToBuild[containerImage] = imageBuildSpec
 }
 
+func (environmemt *ValidatorEnvironment) AppendImageToPullWithAuth(containerImage string, registrySpec *image_registry_spec.ImageRegistrySpec) {
+	environmemt.imagesToPullWithAuth[containerImage] = registrySpec
+}
+
 func (environment *ValidatorEnvironment) GetNumberOfContainerImagesToProcess() uint32 {
```

Comment:
you've gotta update this to account for `environment.imagesToPullWithAuth`

Reply:
good idea